RealmSwift/SwiftUI.swift

```diff
@@ -0,0 +1,599 @@
+     */
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *)
+    public init(wrappedValue: T) where T: ObjectKeyIdentifiable {
+        self._storage = StateObject(wrappedValue: ObservableStorage(wrappedValue))
+        defaultValue = T()
+    }
+}
+
+// MARK: ObservedResults
+
+/// A property wrapper type that retrieves results from a Realm.
+///
+/// The results use the realm configuration provided by
+/// the environment value `EnvironmentValues/realmConfiguration`.
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+@propertyWrapper public struct ObservedResults<ResultType>: DynamicProperty, BoundCollection where ResultType: Object & ObjectKeyIdentifiable {
+    private class Storage: ObservableStorage<Results<ResultType>> {
+        private func didSet() {
+            /// A base value to reset the state of the query if a user reassigns the `filter` or `sortDescriptor`
+            value = try! Realm(configuration: configuration ?? Realm.Configuration.defaultConfiguration).objects(ResultType.self)
+
+            if let sortDescriptor = sortDescriptor {
+                value = value.sorted(byKeyPath: sortDescriptor.keyPath, ascending: sortDescriptor.ascending)
+            }
+            if let filter = filter {
+                value = value.filter(filter)
+            }
+        }
+
+        var sortDescriptor: SortDescriptor? {
+            didSet {
+                didSet()
+            }
+        }
+
+        var filter: NSPredicate? {
+            didSet {
+                didSet()
+            }
+        }
+
+        var configuration: Realm.Configuration? {
+            didSet {
+                didSet()
+            }
+        }
+    }
+
+    @Environment(\.realmConfiguration) var configuration
+    @ObservedObject private var storage = Storage(Results(RLMResults.emptyDetached()))
+    /// :nodoc:
+    @State public var filter: NSPredicate? {
+        willSet {
+            storage.filter = newValue
+        }
+    }
+    /// :nodoc:
+    @State public var sortDescriptor: SortDescriptor? {
+        willSet {
+            storage.sortDescriptor = newValue
+        }
+    }
+    /// :nodoc:
+    public var wrappedValue: Results<ResultType> {
+        storage.configuration != nil ? storage.value.freeze() : storage.value
+    }
+    /// :nodoc:
+    public var projectedValue: Self {
+        return self
+    }
+    /// :nodoc:
+    public init(_ type: ResultType.Type,
+                configuration: Realm.Configuration? = nil,
+                filter: NSPredicate? = nil,
+                sortDescriptor: SortDescriptor? = nil) {
+        self.storage.configuration = configuration
+        self.filter = filter
+        self.sortDescriptor = sortDescriptor
+    }
+
+    public mutating func update() {
+        // When the view updates, it will inject the @Environment
+        // into the propertyWrapper
+        if storage.configuration == nil ||
```

Comment:
Making `Realm.Configuration` conform to Equatable and putting this logic there seems better.